crates/services/p2p/src/service.rs

```diff
@@ -485,12 +498,87 @@ impl<P: TaskP2PService, V, B: Broadcast, T> Task<P, V, B, T> {
     }
 }
 
+struct CachedView {
+    sealed_block_headers: DashMap<Range<u32>, Vec<SealedBlockHeader>>,
+    transactions_on_blocks: DashMap<Range<u32>, Vec<Transactions>>,
+    metrics: bool,
+}
+
+impl CachedView {
+    fn new(metrics: bool) -> Self {
+        Self {
+            sealed_block_headers: DashMap::new(),
+            transactions_on_blocks: DashMap::new(),
+            metrics,
+        }
+    }
```

Comment:
we probably want to also support sub-ranges or even partial ranges here, but can be in the future :)